src/raft/raw_node.rs

```diff
@@ -126,9 +128,11 @@ impl Ready {
     ) -> Ready {
         let mut rd = Ready {
             entries: raft.raft_log.unstable_entries().unwrap_or(&[]).to_vec(),
-            messages: raft.msgs.drain(..).collect(),
             ..Default::default()
         };
+        if !raft.msgs.is_empty() {
+            mem::swap(&mut raft.msgs, &mut rd.messages);
```

Comment:
any benchmark here?

Reply:
It's very obvious that `mem::swap` does not copy the heap values however `drain.collect` will allocate a new buffer then copy all the content from original.

For a 100 length u64 vector:
test bench_drain_collect ... bench:         105 ns/iter (+/- 2)
test bench_swap          ... bench:           2 ns/iter (+/- 0)